crates/ra_hir/src/db.rs

```diff
@@ -76,6 +76,9 @@ pub trait HirDatabase: SourceDatabase + AsRef<HirInterner> {
     #[salsa::invoke(crate::impl_block::impls_in_module)]
     fn impls_in_module(&self, module: Module) -> Arc<ModuleImplBlocks>;
 
+    #[salsa::invoke(crate::impl_block::impls_in_module_source_map_query)]
+    fn impls_in_module_source_map(&self, module: Module) -> Arc<ImplSourceMap>;
```

Comment:
So we need three queries here I suppose: 

```
fn impls_in_module_with_source_map() -> (Arc<ModuleImplBlocks>, Arc<ImplSourceMap>);
fn impls_in_module() -> Arc<ModuleImplBlocks>;
fn impls_in_module_source_map() -> Arc<ImplSourceMap>;
```

The last two queries should have a trivial impl: `.0` and `.1`